assets/src/scripts/monitoring-location/store/statistics-data.js

```diff
@@ -16,6 +20,31 @@ const setMedianStats = function(statisticsData) {
     };
 };
 
+/*
+*  Helper function that takes a group of statistics and checks if any of those are temperatures in celsius. If found,
+* the function will convert to Fahrenheit and add the new temperatures to the statistics group.
+* @param {Object} The median statistics
+* @return {Object} The median statistics, possibly with converted temperatures added
+*/
+const convertCelsiusToFahrenheitAndAddToStats = function(stats) {
+    Object.entries(stats).forEach(stat => {
+        if (config.TEMPERATURE_PARAMETERS.celsius.includes(stat[0])) {
+            const convertedStat = cloneDeep(stat[1]);
+
+            Object.entries(convertedStat).forEach(statEntry => {
+                statEntry[1].forEach(detail => {
+                    detail.p50_va = convertCelsiusToFahrenheit(detail.p50_va);
+                });
+            });
+
+            // add the new key value pair to the stats object
+            stats[`${stat[0]}${config.CALCULATED_TEMPERATURE_VARIABLE_CODE}`] = convertedStat;
+        }
+    });
+
+    return stats;
```

Comment:
This has the same problem as the convert function, it is returning data that is mutable. let's always do the deep clone and then return a brand new object.

Reply:
True - repaired